herddb-core/src/main/java/herddb/model/Table.java

```diff
@@ -440,11 +440,8 @@ public Table build() {
                 if (pk == null) {
                     throw new IllegalArgumentException("column " + pkColumn + " is not defined in table");
                 }
-                if (pk.type != ColumnTypes.STRING
-                        && pk.type != ColumnTypes.LONG
-                        && pk.type != ColumnTypes.INTEGER
-                        && pk.type != ColumnTypes.TIMESTAMP) {
-                    throw new IllegalArgumentException("primary key " + pkColumn + " must be a string or long or integer or timestamp");
+                if (!validatePrimaryKeyDataType(pk.type)) {
+                    throw new IllegalArgumentException("primary key " +pkColumn+ " must be a non null string or long or integer or timestamp");
```

Comment:
you can leave the old message, it is implicit that PKs are not null

Reply:
👍